src/test/java/com/example/bank/userRegister/UserServiceTest.java

```diff
@@ -0,0 +1,157 @@
+import static com.example.bank.userRegister.UserServiceUtils.*;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+//@SpringBootTest - komentarz dla mnie  nie używać tej adnotacji; ładuje cały kontekst aplikacji, czyli długo trwa!!
+
+/**
+ * Czy testy mają sprawdzać NULL i empty skoro jest walidacja ??
+ * I czy w ogóle jakoś obsługuje się tego NULL'a, bo jeżeli przekażemy go do metod CRUDowych,
+ * to leci NullPointerException.
+ * Jak na razie dane do testów typu EMPTY lub NULL są zakomentowane.
+ */
+@ExtendWith(MockitoExtension.class)
+class UserServiceTest {
+    @Mock
+    private UserRepository userRepository;
+    @InjectMocks
+    private UserService userService;
+
+    @DisplayName("A parameterized test of registerUser() method")
+    @ParameterizedTest
+    @MethodSource("userOver18YearsOldSource")
+    void shouldReturnUserWhenUserIsBetween18And119YearsOld(User user, UserRequest userRequest) {
+        when(userRepository.save(user)).thenReturn(user);
+        User resultUser = userService.registerUser(userRequest);
+        Assertions.assertEquals(resultUser, user);
+    }
+
+    @DisplayName("A parameterized test of registerUser() method")
+    @ParameterizedTest
+    @MethodSource("userRequestBelow18YearsOldAndNullAndEmptySource")
+    void shouldReturnIllegalArgumentExceptionWhenAgeIsBetween18And119(UserRequest userRequest) {
+        Assertions.assertThrows(IllegalArgumentException.class, () -> {
+            userService.registerUser(userRequest);
+        }, "User has to be adult.");
+    }
+
+    @DisplayName("A parameterized test of getUser() method")
+    @ParameterizedTest
+    @MethodSource("userOver18YearsOldSource")
+    void shouldGetUserAndReturnUserFromId3(User user) throws UserNotFoundException {
+        Long id = user.getId();
+        when(userRepository.findById(id)).thenReturn(Optional.of(user));
+        User resultUser = userService.getUser(id);
+        Assertions.assertEquals(resultUser, user);
+    }
+
+    @DisplayName("A parameterized test of getUser() method")
+    @ParameterizedTest
+    @MethodSource("userSource")
+    void shouldReturnUserExceptionWhenThereIsNoUserWithId(User user) {
+        Long id = user.getId();
+        when(userRepository.findById(id)).thenReturn(Optional.empty());
+        Assertions.assertThrows(UserNotFoundException.class, () -> {
+            userService.getUser(id);
+        });
+    }
+
+    @DisplayName("A parameterized test of updateUser() method")
+    @ParameterizedTest
+    @MethodSource("userOver18YearsOldUpdateSource")
+    void shouldReturnUpdatedUser(User user, UserRequest userRequest, User updatedUser) throws UserNotFoundException {
+        Long id = user.getId();
+        when(userRepository.findById(id)).thenReturn(Optional.of(user));
+        when(userRepository.save(user)).thenReturn(user);
+        User resultUser = userService.updateUser(userRequest, user.getId());
```

Comment:
Jakos srednio mi sie podoba kwestia ze sa 2 parametry w tej metodzie. Bo to glupio wyglada ze masz userRequest i jeszcze dopychasz ID zamiast miec je od razu w obiekcie requesta

Reply:
Skorygowano w userReqest i testach.